frontend/src/components/TaskMapTask.tsx

```diff
@@ -0,0 +1,160 @@
+    taskSelector(taskId),
+    shallowEqual,
+  );
+
+  const selectTask = (e: MouseEvent) => {
+    e.stopPropagation();
+    setSelectedTask(selected ? undefined : task);
+  };
+
+  if (!task) return null;
+  return (
+    <div
+      role="button"
+      tabIndex={0}
+      onClick={selectTask}
+      onKeyPress={selectTask}
+      className={classes(css.singleTask, {
+        [css.selectedTask]: selected,
+        [css.dragging]: snapshot.isDragging,
+        [css.draggingOutside]: !snapshot.draggingOver,
+        [css.loading]: disableDragging && !snapshot.isDragging,
+      })}
+      ref={provided.innerRef}
+      {...provided.draggableProps}
+      {...provided.dragHandleProps}
+    >
+      {/* drag-n-drop is blocked for interactive elements such as buttons */}
+      <button type="button">
+        <Handle
+          className={classes(css.leftHandle, {
+            [css.filledLeftHandle]: toChecked,
+            [css.dragging]: snapshot.isDragging,
+          })}
+          id={`to-${task.id}`}
+          type="target"
+          position={Position.Left}
+        />
+      </button>
+      {task.completed && <DoneAllIcon className={classes(css.doneIcon)} />}
+      {task.name}
+      <div className={classes(css.taskRatingTexts)}>
+        <TaskRatingsText
+          task={task}
+          selected={selected}
+          largeIcons
+          dragging={snapshot.isDragging}
+        />
+      </div>
+      <button type="button">
+        <Handle
+          className={classes(css.rightHandle, {
+            [css.filledRightHandle]: fromChecked,
+            [css.dragging]: snapshot.isDragging,
+          })}
+          id={`from-${task.id}`}
+          type="source"
+          position={Position.Right}
+        />
+      </button>
+    </div>
+  );
+};
+
+export const DraggableSingleTask: FC<
+  TaskProps & {
+    index: number;
+  }
+> = ({
+  taskId,
+  setSelectedTask,
+  selected,
+  index,
+  toChecked,
+  fromChecked,
+  disableDragging,
+}) => (
+  <Draggable
+    key={taskId}
+    draggableId={`${taskId}`}
+    index={index}
+    isDragDisabled={disableDragging}
+  >
+    {(provided, snapshot) => {
+      if (snapshot.isDragging)
+        return ReactDOM.createPortal(
+          <SingleTask
+            taskId={taskId}
+            selected={selected}
+            setSelectedTask={setSelectedTask}
+            snapshot={snapshot}
+            provided={provided}
+            toChecked={toChecked}
+            fromChecked={fromChecked}
+            disableDragging={disableDragging}
+          />,
+          document.getElementById('taskmap')!,
+        );
+      return (
+        <SingleTask
+          taskId={taskId}
+          selected={selected}
+          setSelectedTask={setSelectedTask}
+          snapshot={snapshot}
+          provided={provided}
+          toChecked={toChecked}
+          fromChecked={fromChecked}
+          disableDragging={disableDragging}
+        />
+      );
```

Comment:
```suggestion
      const element = (
        <SingleTask
          taskId={taskId}
          snapshot={snapshot}
          provided={provided}
          disableDragging={disableDragging}
          {...rest}
        />
      );
      return snapshot.isDragging
        ? ReactDOM.createPortal(element, document.getElementById('taskmap')!)
        : element;
```

Might be worth deduplicating the element construction, as it has quite a few props

The props that are only forwarded could also be passed without explicitly naming each